src/constraints/joints/distance.rs

```diff
@@ -0,0 +1,194 @@
+//! [`DistanceJoint`] component.
+
+use crate::prelude::*;
+use bevy::prelude::*;
+
+/// A distance joint prevents relative translation of the attached bodies away from its rest length while allowing rotation around all axes.
```

Comment:
Just to make this a bit clearer and to make it more distinct from the spherical joint, I would have something like this:

```suggestion
/// A distance joint keeps the attached bodies at a certain distance from each other while while allowing rotation around all axes.
```